pkg/pxc/app/statefulset/node.go

```diff
@@ -142,148 +183,70 @@ func (c *Node) AppContainer(spec *api.PodSpec, secrets string, cr *api.PerconaXt
-			Name:      "auto-config",
-			MountPath: "/etc/my.cnf.d",
+			Name:      "hookscript",
+			MountPath: "/opt/percona/hookscript",
 		})
 	}
 
-	if cr.CompareVersionWith("1.4.0") >= 0 {
-		appc.VolumeMounts = append(appc.VolumeMounts, corev1.VolumeMount{
-			Name:      VaultSecretVolumeName,
-			MountPath: "/etc/mysql/vault-keyring-secret",
-		})
+	if cr.Spec.LogCollector != nil && cr.Spec.LogCollector.Enabled {
+		appc.Env = append(appc.Env, []corev1.EnvVar{
+			{
+				Name:  "LOG_DATA_DIR",
+				Value: "/var/lib/mysql",
+			},
+			{
+				Name:  "IS_LOGCOLLECTOR",
+				Value: "yes",
+			},
+		}...)
 	}
 
-	if cr.CompareVersionWith("1.5.0") >= 0 {
-		appc.Args = []string{"mysqld"}
-		appc.Command = []string{"/var/lib/mysql/pxc-entrypoint.sh"}
-		appc.Env = append(appc.Env, corev1.EnvVar{
+	appc.Env = append(appc.Env, []corev1.EnvVar{
+		{
+			Name:  "CLUSTER_HASH",
+			Value: serverIDHashStr,
+		},
+		{
 			Name: "OPERATOR_ADMIN_PASSWORD",
 			ValueFrom: &corev1.EnvVarSource{
 				SecretKeyRef: app.SecretKeySelector(secrets, users.Operator),
 			},
-		})
-		if cr.CompareVersionWith("1.11.0") >= 0 && cr.Spec.PXC != nil && cr.Spec.PXC.HookScript != "" {
-			appc.VolumeMounts = append(appc.VolumeMounts, corev1.VolumeMount{
-				Name:      "hookscript",
-				MountPath: "/opt/percona/hookscript",
-			})
-		}
-	}
-
-	if cr.CompareVersionWith("1.6.0") >= 0 {
-		appc.Ports = append(
-			appc.Ports,
-			corev1.ContainerPort{
-				ContainerPort: 33062,
-				Name:          "mysql-admin",
-			},
-		)
-		appc.ReadinessProbe.Exec.Command = []string{"/var/lib/mysql/readiness-check.sh"}
-		appc.LivenessProbe.Exec.Command = []string{"/var/lib/mysql/liveness-check.sh"}
-	}
-
-	if cr.CompareVersionWith("1.9.0") >= 0 {
-		appc.Ports = append(
-			appc.Ports,
-			corev1.ContainerPort{
-				ContainerPort: 33060,
-				Name:          "mysqlx",
-			},
-		)
+		},
+		{
+			Name:  "LIVENESS_CHECK_TIMEOUT",
+			Value: fmt.Sprint(spec.LivenessProbes.TimeoutSeconds),
+		},
+		{
+			Name:  "READINESS_CHECK_TIMEOUT",
+			Value: fmt.Sprint(spec.ReadinessProbes.TimeoutSeconds),
+		},
+	}...)
 
-		appc.LivenessProbe = &spec.LivenessProbes
-		appc.ReadinessProbe = &spec.ReadinessProbes
-		appc.ReadinessProbe.Exec = &corev1.ExecAction{
-			Command: []string{"/var/lib/mysql/readiness-check.sh"},
-		}
-		appc.LivenessProbe.Exec = &corev1.ExecAction{
-			Command: []string{"/var/lib/mysql/liveness-check.sh"},
-		}
-		probsEnvs := []corev1.EnvVar{
-			{
-				Name:  "LIVENESS_CHECK_TIMEOUT",
-				Value: fmt.Sprint(spec.LivenessProbes.TimeoutSeconds),
-			},
-			{
-				Name:  "READINESS_CHECK_TIMEOUT",
-				Value: fmt.Sprint(spec.ReadinessProbes.TimeoutSeconds),
-			},
+	if cr.CompareVersionWith("1.13.0") >= 0 {
+		plugin := "caching_sha2_password"
+		if cr.Spec.ProxySQLEnabled() {
+			plugin = "mysql_native_password"
 		}
-		appc.Env = append(appc.Env, probsEnvs...)
+		appc.Env = append(appc.Env, corev1.EnvVar{
+			Name:  "DEFAULT_AUTHENTICATION_PLUGIN",
```

Comment:
@egegunes  @dmitriy-kostiuk We will have PXC pods restart in case of switching from ProxySQL to Haproxy. We need to mention it in our doc and release notes.